text/0000-add-geo-processing-abilities-to-skylark.md

```diff
@@ -0,0 +1,535 @@
+- What is the impact of not doing this? -->
+
+
+
+# Prior art
+[prior-art]: #prior-art
+
+- [PostGIS Functions](https://postgis.net/docs/reference.html#Geometry_Editors) : A breakdown of geospatial functions that PostGIS uses (this is probably the most comprehensive list of these kinds of functions)
+- [golang-geo](https://github.com/kellydunn/golang-geo) : A geospatial library for Go
+- [https://github.com/golang/geo ](https://github.com/golang/geo) : Googles S2 library implement in Go
+- [Shapley](https://pypi.org/project/Shapely/) : One of the core python libraries for geometries, main inspiration for the API here.
+- [GeoPandas](http://geopandas.org/) : DataFrame implementation that has geospatial functionality.
+
+<!-- Discuss prior art, both the good and the bad, in relation to this proposal.
+A few examples of what this can include are:
+
+- Does this feature exist in other places and what experience have their community had?
+- For community proposals: Is this done by some other community and what were their experiences with it?
+- For other teams: What lessons can we learn from what other communities have done here?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other projects, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
+
+Note that while precedent set by other projects is some motivation, it does not on its own motivate an RFC.
+Please also take into consideration that Qri sometimes intentionally diverges from other projects. -->
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+There are a bunch of features that this proposal doesn't include, is it valid to not include these in the first pass:
+- Projections : Geospatial data can be projected in to many different local coordinate systems which are 
+useful in different contexts. We aren't including this in this first implementation but if it's important we could include it
+- Geometry modification methods : Translate, Rotate etc functions 
+
+Other questions:
+
+- Do we want this to help support the visualization part of Qri when data is geospatial?
+- What volumes of data should this be looking to support? Do we need to be building and somehow persisting with the dataset geospatial indexes that allow for faster processing / querying? 
+- Support for other geospatial data formats : Qri currently supports csv, json and cbor formats. JSON can be used to work with GeoJSON but other geospatial formats like KML, Geopackage and  Shape Files to name a few are common. Would these potentially be supported by the cbor format? If not what is the scope for adding formats like these?
```

Comment:
Yeah that's definitely an open question, would love to revisit this after we've landed support. In my head this conversation dovetails well with a long-term chat about plugin support.